phoenix-ios/phoenix-ios/views/HomeView.swift

```diff
@@ -643,54 +643,144 @@ fileprivate struct PaymentCell : View, ViewName {
+		)
+		.onReceive(syncManager.statePublisher) {
+			syncManagerStateChanged($0)
+		}
+		.onReceive(syncManager.pendingSettingsPublisher) {
+			syncManagerPendingSettingsChanged($0)
+		}
+	}
+	
+	@ViewBuilder
+	var buttonContent: some View {
+		
+		let connectionStatus = connectionsMonitor.connections.global
+		if connectionStatus == .closed {
+			HStack(alignment: .firstTextBaseline, spacing: 3) {
+				Image(systemName: "bolt.slash.fill")
+					.imageScale(.large)
+				Text(NSLocalizedString("Offline", comment: "Connection state"))
+					.padding(.trailing, 6)
+			}
+			.font(.caption2)
+		}
+		else if connectionStatus == .establishing {
+			HStack(alignment: .firstTextBaseline, spacing: 3) {
+				Image(systemName: "bolt.slash")
+					.imageScale(.large)
+				Text(NSLocalizedString("Connecting...", comment: "Connection state"))
+					.padding(.trailing, 6)
+			}
+			.font(.caption2)
+		} else /* .established */ {
+			
+			if pendingSettings != nil {
+				// The user enabled/disabled cloud sync.
+				// We are using a 30 second delay before we start operating on the user's decision.
+				// Just in-case it was an accidental change, or the user changes his/her mind.
+				Image(systemName: "hourglass")
+					.imageScale(.large)
+					.font(.caption2)
+					.squareFrame()
+			} else {
+				let (isSyncing, isWaiting, isError) = buttonizeSyncStatus()
+				if isSyncing {
+					Image(systemName: "icloud")
+						.imageScale(.large)
 						.font(.caption2)
+						.squareFrame()
+				} else if isWaiting {
+					Image(systemName: "hourglass")
+						.imageScale(.large)
+						.font(.caption2)
+						.squareFrame()
+				} else if isError {
+					Image(systemName: "exclamationmark.triangle")
+						.imageScale(.large)
+						.font(.caption2)
+						.squareFrame()
+				} else {
+					// Everything is good: connected + {synced|disabled|initializing}
+					Image(systemName: "bolt.fill")
+						.imageScale(.large)
+						.font(.caption2)
+						.squareFrame()
 				}
 			}
-			.buttonStyle(PlainButtonStyle())
-			.padding([.leading, .top, .bottom], 5)
-			.padding([.trailing], 10)
-			.background(Color.buttonFill)
-			.cornerRadius(30)
-			.overlay(
-				RoundedRectangle(cornerRadius: 30)
-					.stroke(Color.borderColor, lineWidth: 1)
-			)
-			.opacity(dimStatus ? 0.2 : 1.0)
-			.isHidden(status == Lightning_kmpConnection.established)
 		}
-		.onAppear {
-			DispatchQueue.main.async {
-				withAnimation(Animation.linear(duration: 1.0).repeatForever()) {
-					self.dimStatus.toggle()
+	}
+	
+	func buttonizeSyncStatus() -> (Bool, Bool, Bool) {
+		
+		var isSyncing = false
+		var isWaiting = false
+		var isError = false
+		
+		switch syncState {
+			case .initializing: break
+			case .updatingCloud: isSyncing = true
+			case .downloading: isSyncing = true
+			case .uploading: isSyncing = true
+			case .waiting(let details):
+				switch details.kind {
+					case .forInternet: break
+					case .forCloudCredentials: isError = true
```

Comment:
If the device is not connected to iCloud, we should not display an error, it gives the feeling that the wallet does not work well. Instead this case should be handled as the disabled case.

Reply:
Done in fe93b06